tgui/packages/tgui/interfaces/TeleConsole.tsx

```diff
@@ -0,0 +1,196 @@
+            />
+            <Button
+              icon="forward"
+              onClick={() => act('setX', { value: xtarget + 10 })}
+            />
+          </Box>
+          <Box>
+            {"Y: "}
+            <Button
+              icon="backward"
+              onClick={() => act('setY', { value: ytarget - 10 })}
+            />
+            <Button
+              icon="caret-left"
+              onClick={() => act('setY', { value: ytarget - 1 })}
+            />
+            <Button.Input
+              content={ytarget}
+              onCommit={(e, value) => act('setY', { value: value })}
+            />
+            <Button
+              icon="caret-right"
+              onClick={() => act('setY', { value: ytarget + 1 })}
+            />
+            <Button
+              icon="forward"
+              onClick={() => act('setY', { value: ytarget + 10 })}
+            />
+          </Box>
+          <Box>
+            {"Z: "}
+            <Button
+              icon="caret-left"
+              onClick={() => act('setZ', { value: ztarget - 1 })}
+            />
+            <Button.Input
+              content={ztarget}
+              onCommit={(e, value) => act('setZ', { value: value })}
+            />
+            <Button
+              icon="caret-right"
+              onClick={() => act('setZ', { value: ztarget + 1 })}
+            />
+          </Box>
+        </Section>
+        <Section width="80%" mx="auto">
+          <Box>
+            <Button
+              color="green"
+              icon="sign-out-alt"
+              onClick={() => act("send")}
+              content="Send"
+              disabled={!host_id}
+            />
+            <Button
+              color="green"
+              icon="sign-in-alt"
+              onClick={() => act("receive")}
+              content="Recieve"
+              disabled={!host_id}
+            />
+            <Button
+              color="green"
+              onClick={() => act("portal")}
+              disabled={!host_id}
+            >
+              <Icon name="ring" rotation={90} />Toggle Portal
+            </Button>
+          </Box>
+          <Button
+            color="green"
+            icon="magnifying-glass"
+            onClick={() => act("scan")}
+            content="Scan"
+            disabled={!host_id}
+          />
+        </Section>
+        <Section width="80%" mx="auto">
+          {"Bookmarks: "}
+          <Button
+            color="green"
+            icon="add"
+            onClick={() => act("addbookmark")}
+            disabled={bookmarks.length >= max_bookmarks}
+          />
+          {bookmarks.map(mark => {
+            return (
+              <Box key={mark["ref"]}>
+                <Button
+                  icon="bookmark"
+                  onClick={() => act("restorebookmark", { value: mark["ref"] })}
+                  content={mark["name"]}
+                />
+                {mark["xyz"]}
```

Comment:
Given how you're using this data, why not pass up as three fields `x`, `y`, `z` which are just numbers then format however you want here?

Reply:
The reason i did this is because i didn't feel like figuring out template strings. Should do this the proper way